src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1705,6 +1705,11 @@ RouterJob(Query *originalQuery, PlannerRestrictionContext *plannerRestrictionCon
 												 MODIFY_TASK,
 												 requiresMasterEvaluation);
 	}
+	else if (shardId == INVALID_SHARD_ID)
+	{
+		/* modification that prunes to 0 shards */
```

Comment:
Given that the `if/else` block is not very simple, and can change in  future,  maybe add `Assert (originalQuery->commandType != CMD_SELECT)`,  because in `SELECT` case it is where `intermediate result only` router queries.